app/src/main/java/com/xinto/opencord/util/FlowUtil.kt

```diff
@@ -0,0 +1,11 @@
+package com.xinto.opencord.util
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+fun <T> Flow<T>.collectIn(scope: CoroutineScope, action: suspend (T) -> Unit): Job {
```

Comment:
Just a comment for the future, we could probably use context receivers as coroutine scope here and default the `scope` parameter to `this`

Reply:
we wouldn't need to explicitly provide scope and instead use the one in current `launch {}` context by default

Reply:
just as I said, it's just for future, nothing else